stack/source.go

```diff
@@ -189,6 +189,10 @@ func fieldToType(f *ast.Field) (string, bool) {
 		return arg.Sel.Name, false
 	case *ast.StarExpr:
 		return "*" + name(arg.X), false
+	case *ast.MapType:
+		return fmt.Sprintf("map[%s]%s", arg.Key.(*ast.Ident).Name, arg.Value.(*ast.Ident).Name), false
+	case *ast.ChanType:
+		return fmt.Sprintf("chan %s", arg.Value.(*ast.Ident).Name), false
```

Comment:
In both ChanType and MapType, I'm wondering what happens if the type is something complex. An example would be a func type without a name, e.g.  `chan func()`. Can you try it? Another thing that may not work as is is pointers, so I think you need to recurse.

Reply:
You were right – `chan func()` was causing it to panic. I've addressed complex types (incl. pointers) in the latest commit.